src/diffusers/pipelines/flux/pipeline_flux.py

```diff
@@ -73,11 +74,24 @@
 
 def calculate_shift(
     image_seq_len,
-    base_seq_len: int = 256,
-    max_seq_len: int = 4096,
-    base_shift: float = 0.5,
-    max_shift: float = 1.16,
+    base_seq_len: Optional[int] = 256,
+    max_seq_len: Optional[int] = 4096,
+    base_shift: Optional[float] = 0.5,
+    max_shift: Optional[float] = 1.16,
+    scheduler: Optional[FlowMatchEulerDiscreteScheduler] = None,
 ):
+    if base_seq_len or max_seq_len or base_shift or max_shift or scheduler is None:
+        deprecation_message = "Pass `scheduler` to `calculate_shift`."
+        deprecate(
+            "calculate_shift scheduler",
+            "1.0.0",
+            deprecation_message,
+            standard_warn=False,
+        )
+    base_seq_len = base_seq_len or scheduler.config.get("base_image_seq_len", 256)
+    max_seq_len = max_seq_len or scheduler.config.get("max_image_seq_len", 4096)
+    base_shift = base_shift or scheduler.config.get("base_shift", 0.5)
+    max_shift = max_shift or scheduler.config.get("max_shift", 1.16)
```

Comment:
Thanks for the contribution @vladmandic, it will be fun to try to those scheduler changes with Flux.

As you mentioned CI, I've ran `make fix-copies` to demonstrate. Every occurrence of `# Copied from diffusers.pipelines.flux.pipeline_flux.calculate_shift` is updated, this highlights the effects in our own codebase, we'll need to update the call of `calculate_shift` in every pipeline and there can be effects downstream, so I've pushed this version which adds a deprecation message. We'll run `make fix-copies` again when the changes here are finalised.

Reply:
thanks!